src/hooks/usePermalink.ts

```diff
@@ -37,13 +35,38 @@ interface Args {
 }
 
 interface HookResult {
-    /** Avatar of the permalinked resource. */
-    avatar: ReactElement | null;
-    /** Displayable text of the permalink resource. Can for instance be a user or room name. */
+    /**
+     * Room member of a user mention permalink.
+     * null for other links, if the profile was not found or not yet loaded.
+     * This can change, for instance, from null to a RoomMember after the profile lookup completed.
+     */
+    member: RoomMember | null;
+    /**
+     * Displayable text of the permalink resource. Can for instance be a user or room name.
+     * null here means that there is nothing to display. Most likely if the URL was no permalink.
+     */
     text: string | null;
-    onClick: ((e: ButtonEvent) => void) | null;
-    /** This can be for instance a user or room Id. */
+    /**
+     * Should be used for click actions on the permalink.
+     * In case of a user permalink, a view profile action is dispatched.
+     */
+    onClick: (e: ButtonEvent) => void;
+    /**
+     * This can be for instance a user or room Id.
+     * null here means that the resource cannot be detected. Most likely if the URL was no permalink.
+     */
     resourceId: string | null;
+    /**
+     * Target room of the permalink:
+     * For an @room, this is the room where the permalink should be displayed.
```

Comment:
I am somewhat surprised to discover that this code is involved with `@room` mentions, given (afaik) we don't use permalinks for them?  But that's a problem for another day. In the meantime, a small clarification:

```suggestion
     * For an @room mention, this is the room where the permalink should be displayed.
```